WeatherApp/ViewControllers/WeatherViewController.swift

```diff
@@ -9,27 +9,67 @@
 import UIKit
 
 class WeatherViewController: UIViewController {
-
     let tableView = UITableView()
+    let weatherService = WeatherApiService()
+    let cities = ["Zagreb", "Split", "Osijek", "Rijeka"]
+    var weatherData: [WeatherData] = [] {
+        didSet {
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+    }
     
-    override func loadView() {
-        super.loadView()
-        
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        tableView.dataSource = self
+        tableView.delegate = self
+        tableView.register(WeatherTableViewCell.self, forCellReuseIdentifier: "weatherCell")
         setupUI()
         setupConstraints()
+        fillTableView()
     }
-
-
+    
+    func fillTableView() {
+        cities.forEach { city in
+            weatherService.fetchWeather(for: city, completion: { (weatherResult) in
+                self.weatherData.append(weatherResult)
+            })
+        }
+    }
+    
     func setupUI() {
         view.addSubview(tableView)
+        tableView.backgroundColor = .systemBlue
+        tableView.separatorColor = .white
         tableView.translatesAutoresizingMaskIntoConstraints = false
     }
     
     func setupConstraints() {
-        tableView.translatesAutoresizingMaskIntoConstraints = false
         tableView.topAnchor.constraint(equalTo: view.topAnchor).isActive = true
         tableView.leftAnchor.constraint(equalTo: view.leftAnchor).isActive = true
         tableView.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
         tableView.rightAnchor.constraint(equalTo: view.rightAnchor).isActive = true
     }
 }
+
+extension WeatherViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return weatherData.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "weatherCell", for: indexPath) as! WeatherTableViewCell
+        cell.weather = weatherData[indexPath.row]
```

Comment:
[] is not a safe operator because it will throw Error if element does not exist. You should check if `weatherData` has a count larger than `indexPath.row` before assigning value to `cell.weather`